src/rpc/blockchain.cpp

```diff
@@ -1597,6 +1599,236 @@ UniValue savemempool(const JSONRPCRequest& request)
     return NullUniValue;
 }
 
+static std::mutex g_utxosetscan;
+static std::atomic<int> g_scan_progress;
+static std::atomic<bool> g_scan_in_progress;
+static std::atomic<bool> g_should_abourt_scan;
+/** RAII object */
+class CoinsViewScanReserver
+{
+private:
+    bool m_could_reserve;
+public:
+    explicit CoinsViewScanReserver() : m_could_reserve(false) {}
+
+    bool reserve() {
+        std::lock_guard<std::mutex> lock(g_utxosetscan);
+        if (g_scan_in_progress) {
+            return false;
+        }
+        g_scan_in_progress = true;
+        m_could_reserve = true;
+        return true;
+    }
+
+    ~CoinsViewScanReserver() {
+        std::lock_guard<std::mutex> lock(g_utxosetscan);
+        if (m_could_reserve) {
+            g_scan_in_progress = false;
+        }
+    }
+};
+
+UniValue scantxoutset(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() < 1 || request.params.size() > 2)
```

Comment:
`request.params.size() != 2`?

Reply:
Nah. You can also call `scantxoutset status`

Reply:
Commit "Blockchain/RPC: Add scantxoutset method to scan UTXO set"

> Nah. You can also call scantxoutset status

Right.

However should be `request.params.size() > 3` (instead of 2)?